File: src/TaskOTime/TaskOTime.ViewModel/ViewModels/TaskListViewModel.vb
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.ComponentModel
Imports System.Linq
Imports TaskOTime.ViewModel.Base

Namespace ViewModels
    Public Class TaskListViewModel
        Inherits ViewModelBase

        Private _title As String
        Private _subtitle As String

        Public Sub New(title As String, subtitle As String, tasks As IEnumerable(Of TaskItemViewModel))
            _title = title
            _subtitle = subtitle
            Me.Tasks = New ObservableCollection(Of TaskItemViewModel)(tasks)

            For Each taskItem In Me.Tasks
                AddHandler taskItem.PropertyChanged, AddressOf OnTaskItemPropertyChanged
            Next
        End Sub

        Public Property Title As String
            Get
                Return _title
            End Get
            Set(value As String)
                SetProperty(_title, value, NameOf(Title))
            End Set
        End Property

        Public Property Subtitle As String
            Get
                Return _subtitle
            End Get
            Set(value As String)
                SetProperty(_subtitle, value, NameOf(Subtitle))
            End Set
        End Property

        Public ReadOnly Property Tasks As ObservableCollection(Of TaskItemViewModel)

        Public ReadOnly Property OpenTaskCount As Integer
            Get
                Return Tasks.Where(Function(taskItem) taskItem.IsOpen).Count()
            End Get
        End Property

        Public ReadOnly Property DoneTaskCount As Integer
            Get
                Return Tasks.Where(Function(taskItem) taskItem.IsDone).Count()
            End Get
        End Property

        Public ReadOnly Property ProgressSummary As String
            Get
                Return $"{OpenTaskCount} offen · {DoneTaskCount} erledigt"
            End Get
        End Property

        Private Sub OnTaskItemPropertyChanged(sender As Object, e As PropertyChangedEventArgs)
            If e.PropertyName = NameOf(TaskItemViewModel.IsDone) OrElse
               e.PropertyName = NameOf(TaskItemViewModel.IsOpen) Then
                OnPropertyChanged(NameOf(OpenTaskCount))
                OnPropertyChanged(NameOf(DoneTaskCount))
                OnPropertyChanged(NameOf(ProgressSummary))
            End If
        End Sub
    End Class
End Namespace

File: src/TaskOTime/TaskOTime.ViewModel/ViewModels/TaskListViewModel.cs
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.Linq;
using TaskOTime.ViewModel.Base;

namespace TaskOTime.ViewModel.ViewModels
{
    public class TaskListViewModel : ViewModelBase
    {

        private string _title;
        private string _subtitle;

        public TaskListViewModel(string title, string subtitle, IEnumerable<TaskItemViewModel> tasks)
        {
            _title = title;
            _subtitle = subtitle;
            Tasks = new ObservableCollection<TaskItemViewModel>(tasks);

            foreach (var taskItem in Tasks)
                taskItem.PropertyChanged += OnTaskItemPropertyChanged;
        }

        public string Title
        {
            get
            {
                return _title;
            }
            set
            {
                SetProperty(ref _title, value, nameof(Title));
            }
        }

        public string Subtitle
        {
            get
            {
                return _subtitle;
            }
            set
            {
                SetProperty(ref _subtitle, value, nameof(Subtitle));
            }
        }

        public ObservableCollection<TaskItemViewModel> Tasks { get; private set; }

        public int OpenTaskCount
        {
            get
            {
                return Tasks.Where(taskItem => taskItem.IsOpen).Count();
            }
        }

        public int DoneTaskCount
        {
            get
            {
                return Tasks.Where(taskItem => taskItem.IsDone).Count();
            }
        }

        public string ProgressSummary
        {
            get
            {
                return $"{OpenTaskCount} open · {DoneTaskCount} completed";
            }
        }

        private void OnTaskItemPropertyChanged(object sender, PropertyChangedEventArgs e)
        {
            if ((e.PropertyName ?? "") == nameof(TaskItemViewModel.IsDone) || (e.PropertyName ?? "") == nameof(TaskItemViewModel.IsOpen))
            {
                OnPropertyChanged(nameof(OpenTaskCount));
                OnPropertyChanged(nameof(DoneTaskCount));
                OnPropertyChanged(nameof(ProgressSummary));
            }
        }
    }
}